src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1059,6 +1075,27 @@ internal void DeleteModelInternal(List<ModelBase> modelsToDelete)
             OnDeletionComplete(this, EventArgs.Empty);
         }
 
+        internal void UngroupModel(List<ModelBase> modelsToUngroup)
```

Comment:
Are `AnnotationModel`'s not allowed in `CustomNodeWorkspaceModel`?  That is a bad limitation...